### DISCUSSION TITLE
Booting fails after flashing ESP32-WROOM-32D

### DISCUSSION THREAD

**Discussion by jeroenst** [arendst/Tasmota#18211] [Q&A] [answered]
I flashed a ESP32-WROOM-32D chip but it fails to boot:

09:40:23.489 -> rst:0x10 (RTCWDT_RTC_RESET),boot:0x17 (SPI_FAST_FLASH_BOOT)
09:40:23.489 -> flash read err, 1000
09:40:23.489 -> ets_main.c 371 
09:40:23.821 -> ets Jun  8 2016 00:22:57

The command I used was:
python3 esptool.py erase_flash
python3 esptool.py --chip esp32 --baud 921600 --before default_reset --after hard_reset write_flash -z --flash_mode dout --flash_size detect 0x0 ../../Downloads/tasmota32.factory.bin

Flashing succeeds without any errors.

How can I get tasmota running on this device?

**Comment by jeroenst** [comment]
Seems like latest tasmota version 12.4.0 (http://ota.tasmota.com/tasmota32/release/tasmota32.factory.bin) is broken, flashing 12.3.1 (http://ota.tasmota.com/tasmota32/release-12.3.1/) works file.

**Comment by Jason2866** [comment, marked as answer]
Wrong flash command. Do NEVER use flash size and mode commands with a factory image file.
This will work
```
python3 esptool.py write_flash 0x0 ../../Downloads/tasmota32.factory.bin
```

**Reply by jeroenst** [reply]
I got it from the tasmota documentation:

https://tasmota.github.io/docs/ESP32/#flashing

**Comment by Jason2866** [comment]
@jeroenst thx I will fix the docu
